test-utils/src/random.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::random::{Rng, SeedableRng};
+use rand_chacha::ChaChaRng;
+
+#[derive(Debug, Copy, Clone)]
+pub struct Seed(pub u64);
+
+impl Seed {
+    pub fn from_entropy() -> Self {
+        Seed(crypto::random::make_true_rng().gen::<u64>())
+    }
+
+    pub fn from_u64(v: u64) -> Self {
+        Seed(v)
+    }
```

Comment:
I think that `rstest` uses the `From` trait to convert test case values. I.e. if you also implement `From<u64> for Seed`, you might be able to write:
```rust
#[case(1337u64)]
fn blah(seed: Seed) {...}
```
Instead of:

```rust
#[case(Seed::from_u64(1337))]
fn blah(seed: Seed) {...}
```

But it might be better to be explicit, not sure.

Reply:
Unfortunately `rstest` only uses `FromStr` trait. Anyway, I think it's still useful to be able to write like this:
```
#[case("7858")]
fn test...
```